src/js/editor/editor.js

```diff
@@ -571,8 +520,9 @@ class Editor {
     section.renderNode.markDirty();
   }
 
-  reparseSection(section) {
-    this._parser.reparseSection(section, this._renderTree);
+  _reparseCurentSection() {
+    const {headSection:currentSection } = this.cursor.offsets;
```

Comment:
The idea here was to push some complexity down from `reparse` to clarify what it does ("reparse current section", then "remove detached sections"). The specificity of this method also helps to clarify that we only ever need to reparse the section that the cursor is in, which wasn't obvious (to me) before I made this change.

I can change it to accept a section to reparse — perhaps a default arg?
I also notice there is a typo in the method name that I will fix.